### PR TITLE
bump python 3.9+

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Python 3.8 is not live, let's bump...
[endoflife.date/python](https://endoflife.date/python)

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20413.org.readthedocs.build/en/20413/

cc @borda @justusschock @awaelchli

### CODE REVIEW

**Comment by Borda** [comment]
```
FAILED trainer/test_trainer.py::test_model_in_correct_mode_during_stages[ddp_spawn-1] - Failed: Timeout >60.0s
FAILED trainer/test_trainer.py::test_error_handling_all_stages[ddp_spawn] - Failed: Timeout >60.0s
```
crashing on MacOS for PT 2.1

**Comment by t-vi** [comment]
> crashing on MacOS for PT 2.1

should be mostly flaky, but I wonder if we should bump all pt 2.1 to 2.1.2, too.

**Comment by Borda** [comment]
> > crashing on MacOS for PT 2.1
> 
> should be mostly flaky

right, but the failing was quite consistent and happened quite often so we may consider doing a rerun...

> I wonder if we should bump all pt 2.1 to 2.1.2

do you mean but package dependency or just for testing/CI

File: .actions/assistant.py [hunk 1]
```diff
@@ -18,10 +18,11 @@
 import shutil
 import tempfile
 import urllib.request
+from collections.abc import Iterable, Iterator, Sequence
 from itertools import chain
 from os.path import dirname, isfile
 from pathlib import Path
-from typing import Any, Dict, Iterable, Iterator, List, Optional, Sequence, Tuple
+from typing import Any, Optional
```
Comment (t-vi) [inline .actions/assistant.py:25]: Given the amount of changes and expecting a fair number of merge conflicts, I'd recommend merging all other patches immediately ready first and then push this.

**Review by t-vi — APPROVED** [review]
Thank you @Borda 
This is good to have, but let's merge the other things first to minimize the number of conflicts.

**Comment by Borda** [comment]
> This is good to have, but let's merge the other things first to minimize the number of conflicts.

good call, let's hold it as the last one 🦩

File: .actions/assistant.py (hunk 1 above)
Comment (Borda) [reply .actions/assistant.py:25]: @t-vi is it a good time to land it now?

**Pull request merged**